xds/internal/balancer/clusterresolver/resource_resolver_eds.go

```diff
@@ -92,13 +93,39 @@ func (er *edsDiscoveryMechanism) OnError(err error) {
 		return
 	}
 
-	er.topLevelResolver.onError(err)
+	if er.logger.V(2) {
+		er.logger.Infof("EDS discovery mechanism for resource %q reported error: %v", er.nameToWatch, err)
+	}
+
+	// If a previously received good configuration exists, continue to use it.
+	// Else report an empty update that would result in the child policy
+	// reporting TRANSIENT_FAILURE (as there would be no priorities or
+	// localities).
+	er.mu.Lock()
+	if er.update != nil {
+		er.mu.Unlock()
+		return
+	}
+	er.update = &xdsresource.EndpointsUpdate{}
+	er.mu.Unlock()
+
+	er.topLevelResolver.onUpdate()
 }
 
 func (er *edsDiscoveryMechanism) OnResourceDoesNotExist() {
 	if er.stopped.HasFired() {
 		return
 	}
 
-	er.topLevelResolver.onError(xdsresource.NewErrorf(xdsresource.ErrorTypeResourceNotFound, "resource name %q of type Endpoints not found in received response", er.nameToWatch))
+	if er.logger.V(2) {
+		er.logger.Infof("EDS discovery mechanism for resource %q reported resource-does-not-exist error", er.nameToWatch)
+	}
+
+	// Report an empty update that would result in the child policy reporting
```

Comment:
Same here.

Reply:
Done. Same as previous. Rewrote it, but not exactly how you wrote it.